specification/confidentialledger/data-plane/Microsoft.ConfidentialLedger/preview/2024-08-22-preview/confidentialledger.json

```
@@ -0,0 +1,1867 @@
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      }
+    },
+    "/app/userDefinedEndpoints/runTimeOptions": {
+      "get": {
+        "operationId": " GetRuntimeOptions ",
+        "x-ms-examples": {
+          "GetRuntimeOptions": {
+            "$ref": "./examples/ledger/GetRuntimeOptions.json"
+          }
+        },
+        "summary": "Runtime options for user defined endpoints.",
+        "description": "It returns the runtime options",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": " Runtime options for user defined endpoints.",
+            "schema": {
+              "$ref": "#/definitions/JSRuntimeOptions"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      },
+      "patch": {
+        "operationId": "PatchRuntimeOptions",
+        "x-ms-examples": {
+          "PatchRuntimeOptions": {
+            "$ref": "./examples/ledger/PatchRuntimeOptions.json"
+          }
+        },
+        "summary": "Runtime options for user defined endpoints.",
+        "description": "Updates the runtime options.",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/JSRuntimeOptionsParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": " Runtime options for user defined endpoints.",
+            "schema": {
+              "$ref": "#/definitions/JSRuntimeOptions"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "common.json#/definitions/ConfidentialLedgerError"
+            }
+          }
+        }
+      }
+    },
+    "/app/userDefinedEndpoints/modules": {
+      "get": {
+        "operationId": "GetUserDefinedEndpointsModule",
+        "x-ms-examples": {
+          "GetUserDefinedEndpointsModule": {
+            "$ref": "./examples/ledger/GetUserDefinedEndpointsModule.json"
+          }
+        },
+        "summary": "Module for user defined endpoints.",
+        "description": "It gets the module for the user defined endpoint.",
+        "parameters": [
+          {
+            "$ref": "common.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/ModuleParameter"
+          }
```

GET operations may not have request bodies.
```suggestion
          }
```